arrows/core/tests/test_derive_metadata.cxx

```diff
@@ -0,0 +1,127 @@
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+namespace {
+
+kwiver::vital::metadata_vector
+make_metadata()
+{
+  kv::metadata_sptr m1 = std::make_shared< kv::metadata >();
+
+  // Add the double-valued traits
+  std::map< kwiver::vital::vital_metadata_tag, double >
+  double_metadata{ { kv::VITAL_META_PLATFORM_HEADING_ANGLE, 60 },
+    { kv::VITAL_META_PLATFORM_PITCH_ANGLE, 3 },
+    { kv::VITAL_META_PLATFORM_ROLL_ANGLE, 1.2 },
+    { kv::VITAL_META_SENSOR_REL_AZ_ANGLE, 5 },
+    { kv::VITAL_META_SENSOR_REL_EL_ANGLE, 4 },
+    { kv::VITAL_META_SENSOR_REL_ROLL_ANGLE, 5 },
+    { kv::VITAL_META_PLATFORM_ROLL_ANGLE, 0.5 },
+    { kv::VITAL_META_SENSOR_VERTICAL_FOV, 30 },
+    { kv::VITAL_META_SENSOR_HORIZONTAL_FOV, 40 },
+    { kv::VITAL_META_SLANT_RANGE, 300 },
+    { kv::VITAL_META_TARGET_WIDTH, 100 } };
+
+  for( auto const& elem : double_metadata )
+  {
+    auto const& trait = meta_traits.find( elem.first );
+    m1->add( trait.create_metadata_item( elem.second ) );
+  }
+
+  // Add the geo point traits
+  auto const& sensor_loc_trait =
+    meta_traits.find( kv::VITAL_META_SENSOR_LOCATION );
+  m1->add( sensor_loc_trait.create_metadata_item(
+             kv::geo_point( kv::geo_point::geo_3d_point_t{ 0, 0, 12 },
+                            kv::SRID::lat_lon_WGS84 ) ) );
+
+  auto const& frame_center_trait =
+    meta_traits.find( kv::VITAL_META_FRAME_CENTER );
+  m1->add( frame_center_trait.create_metadata_item(
+             kv::geo_point( kv::geo_point::geo_3d_point_t{ 0, 0, 0 },
+                            kv::SRID::lat_lon_WGS84 ) ) );
+
+  // Replicate these values except without the slant range field
+  auto const m2 = std::make_shared< kv::metadata >( m1->deep_copy() );
+  m2->erase( kv::VITAL_META_SLANT_RANGE );
+  return { m1, m2 };
+}
+
+} // end namespace
+
+class derive_metadata : public ::testing::Test
+{
+  void
+  SetUp()
+  {
+    auto meta = make_metadata();
+    size_t const frame_height{ 720 };
+    size_t const frame_width{ 1080 };
+
+    derived_metadata = kwiver::arrows::core::compute_derived_metadata(
+      meta, frame_width, frame_height );
+  }
+
+public:
+  kv::metadata_vector derived_metadata;
+};
+
+// ----------------------------------------------------------------------------
+TEST_F(derive_metadata, compute_derived)
+{
+  kv::metadata_item const& gsd_value = derived_metadata.at( 0 )->find(
+    kv::VITAL_META_AVERAGE_GSD );
+  kv::metadata_item const& vniirs_value = derived_metadata.at( 0 )->find(
+    kv::VITAL_META_VNIIRS );
+  kv::metadata_item const& slant_range_value = derived_metadata.at( 0 )->find(
+    kv::VITAL_META_SLANT_RANGE );
+
+  EXPECT_FLOAT_EQ( gsd_value.as_double(), 0.38253304 );
+  EXPECT_FLOAT_EQ( vniirs_value.as_double(), 5.183558 );
+  EXPECT_FLOAT_EQ( slant_range_value.as_double(), 300 );
```

Comment:
Two issues:
- Use `EXPECT_DOUBLE_EQ`, not `EXPECT_FLOAT_EQ`
- Assertions take arguments as `expected, actual`

```suggestion
  EXPECT_DOUBLE_EQ( 0.38253304, gsd_value.as_double() );
  EXPECT_DOUBLE_EQ( 5.183558, vniirs_value.as_double() );
  EXPECT_DOUBLE_EQ( 300, slant_range_value.as_double() );
```

(Also, below)